packages/library/src/commonMain/kotlin/io/realm/RealmList.kt

```diff
@@ -16,232 +16,31 @@
-) : MutableList<E> by delegate {
-
-    internal val listPtr: NativePointer
-        get() = delegate.listPtr
-
-    /**
-     * Constructs a RealmList in unmanaged mode.
-     */
-    constructor() : this(UnmanagedListDelegate())
-
-    /**
-     * Constructs a RealmList in managed mode. For internal use only.
-     */
-    internal constructor(
-        listPtr: NativePointer,
-        metadata: OperatorMetadata
-    ) : this(ManagedListDelegate(listPtr, metadata))
-
-    /**
-     * Observe changes to a RealmList. Follows the pattern of [Realm.addChangeListener].
-     */
-    fun observe(): Flow<RealmList<E>> = delegate.observe(this)
-
-    internal fun freeze(realm: RealmReference): RealmList<E> = delegate.freeze(realm)
-    internal fun thaw(realm: RealmReference): RealmList<E> = delegate.thaw(realm)
-
-    @Suppress("TooGenericExceptionCaught")
-    internal fun isValid(): Boolean {
-        // FIXME workaround until https://github.com/realm/realm-core/issues/4843 is done
-        try {
-            size
-        } catch (e: RuntimeException) {
-            if (e.message?.lowercase()?.contains("access to invalidated list object") == true) {
-                return false
-            }
-        }
-        return true
-    }
-
-    /**
-     * Metadata needed to correctly instantiate a list operator. For internal use only.
-     */
-    internal data class OperatorMetadata(
-        val clazz: KClass<*>,
-        val mediator: Mediator,
-        val realm: RealmReference
-    )
-
-    /**
-     * Facilitates conversion between Realm Core types and Kotlin types and other Realm-related
-     * checks.
-     */
-    internal class Operator<E>(
-        private val metadata: OperatorMetadata
-    ) {
-
-        /**
-         * Converts the underlying Core type to the correct type expressed in the RealmList.
-         */
-        @Suppress("UNCHECKED_CAST")
-        fun convert(value: Any?): E {
-            if (value == null) {
-                return null as E
-            }
-            return with(metadata) {
-                when (clazz) {
-                    Byte::class -> (value as Long).toByte()
-                    Char::class -> (value as Long).toChar()
-                    Short::class -> (value as Long).toShort()
-                    Int::class -> (value as Long).toInt()
-                    Long::class,
-                    Boolean::class,
-                    Float::class,
-                    Double::class,
-                    String::class -> value
-                    else -> (value as Link).toRealmObject(
-                        clazz as KClass<out RealmObject>,
-                        mediator,
-                        realm
-                    )
-                } as E
-            }
-        }
-    }
+interface RealmList<E> : MutableList<E> {
+    fun observe(): Flow<RealmList<E>>
 }
 
 /**
- * Delegate interface for Realm-specific methods.
+ * Instantiates a [RealmList] in **unmanaged** mode.
```

Comment:
```suggestion
 * Instantiates an **unmanaged** [RealmList].
```

Talking about a "mode" make it seems like you can easily switch it IMO.